### PR TITLE
Weapon stats page improvements

### PR DESCRIPTION
**Pull request by MadmanMartian**
## About The Pull Request

Makes the weapon info page a little more informative 
- Adds a description to most firemodes
- If a firemode doesn't change anything stat-wise, it won't show nulls
- If a firemode changes the projectile, it tells you the projectile's name

Planned:
 - Show damage values of the projectile the weapon will fire.
 - Show what a weapon's gunmods does for the weapon.

## Why It's Good For The Game
More informative.

## Changelog
:cl:
add: The weapon info page has been improved
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/stranger.dm [hunk 1]
```diff
@@ -133,10 +133,10 @@
...
+		list(mode_name="uo4E6SBeGe", mode_desc="c25F2OeGUi", burst=1, projectile_type=/obj/item/projectile/plasma/light, fire_sound='sound/weapons/Taser.ogg', fire_delay=5, move_delay=null, charge_cost=3, icon="stun", projectile_color = "#0000FF"),
+		list(mode_name="0sXYAJGCv4", mode_desc="yQI241FKDh", burst=1, projectile_type=/obj/item/projectile/plasma, fire_sound='sound/weapons/Laser.ogg', fire_delay=10, move_delay=null, charge_cost=6, icon="kill", projectile_color = "#FF0000"),
+		list(mode_name="XhddhrdJkJ", mode_desc="uDsfMdPQkm", burst=1, projectile_type=/obj/item/projectile/plasma/heavy, fire_sound='sound/weapons/pulse.ogg', fire_delay=15, move_delay=null, charge_cost=9, icon="destroy", projectile_color = "#FFFFFF"),
+		list(mode_name="bP6hfnj3Js", mode_desc="AhG8GjobYa", burst=3, projectile_type=/obj/item/projectile/plasma/heavy, fire_sound='sound/weapons/pulse.ogg', fire_delay=5, move_delay=4, charge_cost=11, icon="vaporize", projectile_color = "#FFFFFF", recoil_buildup=3)
```
Comment (TheShown911) [inline code/modules/mob/living/simple_animal/hostile/stranger.dm:139]: The numbers, what do they mean...

File: code/modules/projectiles/guns/energy/shrapnel.dm [hunk 2]
```diff
@@ -19,9 +19,9 @@
 	fire_delay = 12 //Equivalent to a pump then fire time
 	fire_sound = 'sound/weapons/guns/fire/energy_shotgun.ogg'
 	init_firemodes = list(
-		list(mode_name="pellet", projectile_type=/obj/item/projectile/bullet/pellet/shotgun, charge_cost=100, icon="kill"),
-		list(mode_name="stun", projectile_type=/obj/item/projectile/bullet/shotgun/beanbag, charge_cost=25, icon="stun"),
-		list(mode_name="blast", projectile_type=/obj/item/projectile/bullet/shotgun, charge_cost=null, icon="destroy"),
+		list(mode_name="Buckshot", mode_desc="Fires a buckshot synth-shell", projectile_type=/obj/item/projectile/bullet/pellet/shotgun, charge_cost=100, icon="kill"),
+		list(mode_name="Beanbag", mode_desc="Fires a beanbag synth-shell", projectile_type=/obj/item/projectile/bullet/shotgun/beanbag, charge_cost=25, icon="stun"),
+		list(mode_name="blast", mode_desc="Fires a slug synth-shell", projectile_type=/obj/item/projectile/bullet/shotgun, charge_cost=null, icon="destroy"),
```
Comment (TheShown911) [inline code/modules/projectiles/guns/energy/shrapnel.dm:24]: Big B Blast please.

**Review by TheShown911 — APPROVED** [review]
Pictures of UI changes are always appreciated.

File: code/modules/mob/living/simple_animal/hostile/stranger.dm (hunk 1 above)
Comment (MadmanMartian) [reply code/modules/mob/living/simple_animal/hostile/stranger.dm:139]: Random strings, because it's an alien technology.

**Comment by MadmanMartian** [comment]
image

File: code/__DEFINES/weapons.dm [hunk 3]
```diff
@@ -47,19 +47,24 @@
...
-#define FULL_AUTO_400		list(mode_name = "full auto",  mode_type = /datum/firemode/automatic, fire_delay = 1.5, icon="auto", damage_mult_add = -0.1)
-#define FULL_AUTO_600		list(mode_name = "full auto",  mode_type = /datum/firemode/automatic, fire_delay = 1  , icon="auto", damage_mult_add = -0.1)
-#define FULL_AUTO_800		list(mode_name = "full auto",  mode_type = /datum/firemode/automatic, fire_delay = 0.8, icon="auto", damage_mult_add = -0.1)
+#define FULL_AUTO_300		list(mode_name = "full auto",  mode_desc = "300 rounds per minute",   mode_type = /datum/firemode/automatic, fire_delay = 2  , icon="auto", damage_mult_add = -0.1)
+#define FULL_AUTO_400		list(mode_name = "full auto",  mode_desc = "400 rounds per minute",   mode_type = /datum/firemode/automatic, fire_delay = 1.5, icon="auto", damage_mult_add = -0.1)
+#define FULL_AUTO_600		list(mode_name = "full auto",  mode_desc = "600 rounds per minute",   mode_type = /datum/firemode/automatic, fire_delay = 1  , icon="auto", damage_mult_add = -0.1)
+#define FULL_AUTO_800		list(mode_name = "full auto",  mode_desc = "800 rounds per minute",   mode_type = /datum/firemode/automatic, fire_delay = 0.8, icon="auto", damage_mult_add = -0.1)
```
Comment (TheShown911) [inline code/__DEFINES/weapons.dm:53]:
```suggestion
#define FULL_AUTO_800		list(mode_name = "fuller auto",  mode_desc = "800 rounds per minute",   mode_type = /datum/firemode/automatic, fire_delay = 0.8, icon="auto", damage_mult_add = -0.1)
```

Comment (MadmanMartian) [reply code/__DEFINES/weapons.dm:53]: hah, yes

**Review by TorinoFermic — APPROVED** [review]

**Pull request merged**